ci/test-clang-docker.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+set -eu
+
+if [[ ! -d /repo ]]; then
+    echo "[-] This script needs to be run within Docker! Use test-clang.sh instead."
+    exit 1
+fi
+
+CC=/repo/toolchain/bin/clang
+CXX=/repo/toolchain/bin/clang++
+hello_c=/repo/llvm-project/hello.c
+hello_cpp=/repo/llvm-project/hello.cpp
+hello_exe=/tmp/hello
+host_arch=$(uname -m)
+
+# Print clang information, which makes sure it can run
+echo "[+] Testing 'clang --version'"
+"$CC" --version
+
+case "$(source /usr/lib/os-release; echo "$ID")" in
+    arch)
+        musl_cc_flags=(
+            -B /usr/lib/musl/lib     # Scrt1.o, crti.o, crtn.o
+            -I /usr/lib/musl/include # stdio.h
+            -L /usr/lib/musl/lib     # -lc
```

Comment:
What about `--sysroot`? Can `--sysroot=/usr/lib/musl` subsume all of `-B, `-I`, `-L`?

Reply:
I'll test.

Reply:
`--sysroot` works for Arch Linux:

https://github.com/nathanchance/containers/runs/6630314825?check_suite_focus=true

But not Fedora, likely due to `lib64` instead of `lib`?

https://github.com/nathanchance/containers/runs/6630479212?check_suite_focus=true

Reply:
I'll file a todo to see what we can do to simplify this. Basically, if we want to support kernel builds for `make LLVM=1` and not just `make CC=clang` with this (i.e. `HOSTCC=clang`) we might have to add these somehow.

Though perhaps I'm getting ahead of myself, and they can simply just use `--target=x86_64-linux-gnu` and it will work just fine.

Perhaps we need to find the most portable way of finding the expected target triple for a given system. Is it glibc based or musl based? Is `$(CC) -print-target-triple` the most portable way to find out? Is there a way to find out without a compiler installed? IDK

example
`$ clang --target=$($(CC) -print-target-triple`

so that our default implicit target doesn't matter.

Reply:
> Perhaps we need to find the most portable way of finding the expected target triple for a given system.

Right, I think this is the root of the problem: for `HOSTCC=clang`, we don't set a triple, we rely on the implicit default, which is not going to work if we want to support both glibc and musl with this toolchain.

> Is it glibc based or musl based?

I suppose one option would be trying to look at a binary for `GLIBC` symbols but that feels like a massive hack...

We could potentially make this work with a wrapper like musl ships? On my host:

```
$ cat $(which musl-clang)
#!/bin/sh
cc="clang"
libc="/usr/lib/musl"
libc_inc="/usr/lib/musl/include"
libc_lib="/usr/lib/musl/lib"
thisdir="`cd "$(dirname "$0")"; pwd`"

# prevent clang from running the linker (and erroring) on no input.
sflags=
eflags=
for x ; do
    case "$x" in
        -l*) input=1 ;;
        *) input= ;;
    esac
    if test "$input" ; then
        sflags="-l-user-start"
        eflags="-l-user-end"
        break
    fi
done

exec $cc \
    -B"$thisdir" \
    -fuse-ld=musl-clang \
    -static-libgcc \
    -nostdinc \
    --sysroot "$libc" \
    -isystem "$libc_inc" \
    -L-user-start \
    $sflags \
    "$@" \
    $eflags \
    -L"$libc_lib" \
    -L-user-end
```

I suspect that might make the regular kernel build unhappy but maybe not?